CI sporadically fails: Exception: Address not found for network_name=development

### The bug

This happens sporadically in CI.

Example: [this CI log file](https://github.com/oceanprotocol/ocean.py/actions/runs/3782922361/jobs/6431054747)

Every single test fails the same way: "Exception: Address not found for network_name=development. Please check your address_file=/home/runner/.ocean/ocean-contracts/artifacts/address.json."

Note: the failures are not happening locally. Just in remote CI.

### Related

This was an issue before locally as well, as reported in [issue#1174](https://github.com/oceanprotocol/ocean.py/issues/1174), worked on in [PR#1175](https://github.com/oceanprotocol/ocean.py/pull/1175), and merged in [this commit](https://github.com/oceanprotocol/ocean.py/commit/c45e3baeed87511c2ff634963773cb2f6142a332).

Apparently we're not in the clear yet!

### Detailed traceback
```text
==================================== ERRORS ====================================
_________________________ ERROR at setup of test_init __________________________

config = {'ADDRESS_FILE': '/home/runner/.ocean/ocean-contracts/artifacts/address.json', 'DOWNLOADS_PATH': 'consume-downloads', 'METADATA_CACHE_URI': 'http://172.15.0.5:5000/', 'NETWORK_NAME': 'development', ...}

    @pytest.fixture
    def ocean_address(config) -> str:
>       return get_address_of_type(config, "Ocean")

conftest_ganache.py:109: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/opt/hostedtoolcache/Python/3.8.15/x64/lib/python3.8/site-packages/enforce_typing/decorator.py:29: in wrapper
    return func(*args, **kwargs)
ocean_lib/ocean/util.py:20: in get_address_of_type
    addresses = get_contracts_addresses(config_dict)
/opt/hostedtoolcache/Python/3.8.15/x64/lib/python3.8/site-packages/enforce_typing/decorator.py:29: in wrapper
    return func(*args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

config = {'ADDRESS_FILE': '/home/runner/.ocean/ocean-contracts/artifacts/address.json', 'DOWNLOADS_PATH': 'consume-downloads', 'METADATA_CACHE_URI': 'http://1[72](https://github.com/oceanprotocol/ocean.py/actions/runs/3782922361/jobs/6431054747#step:7:73).15.0.5:5000', 'NETWORK_NAME': 'development', ...}

    @enforce_types
    def get_contracts_addresses(config: dict) -> Optional[Dict[str, str]]:
        """Get addresses for given NETWORK_NAME, from info in ADDRESS_FILE"""
        network_name = config["NETWORK_NAME"]
        if network_name == "polygon-test":
            network_name = "mumbai"
    
        if network_name == "polygon-main":
            network_name = "polygon"
        addresses = get_contracts_addresses_all_networks(config)
    
        network_addresses = [val for key, val in addresses.items() if key == network_name]
    
        if not network_addresses:
            address_file = config.get("ADDRESS_FILE")
>           raise Exception(
                f"Address not found for network_name={network_name}."
                f" Please check your address_file={address_file}."
            )
E           Exception: Address not found for network_name=development. Please check your address_file=/home/runner/.ocean/ocean-contracts/artifacts/address.json.

ocean_lib/web3_internal/contract_utils.py:67: Exception
```

--- a/.github/dependabot.yml
+++ b/.github/dependabot.yml
@@ -10,7 +10,9 @@
 
 version: 2
 updates:
-  - package-ecosystem: "" # See documentation for possible values
-    directory: "/" # Location of package manifests
-    schedule:
-      interval: "weekly"
+- package-ecosystem: pip
+  directory: "/"
+  schedule:
+    interval: weekly
+    time: '03:00'
+    timezone: Europe/Berlin

--- a/.github/dependabot.yml
+++ b/.github/dependabot.yml
@@ -1,3 +1,8 @@
+##
+## Copyright 2022 Ocean Protocol Foundation
+## SPDX-License-Identifier: Apache-2.0
+##
+
 # To get started with Dependabot version updates, you'll need to specify which
 # package ecosystems to update and where the package manifests are located.
 # Please see the documentation for all configuration options:

--- a/.github/workflows/libcheck.yml
+++ b/.github/workflows/libcheck.yml
@@ -29,7 +29,7 @@ jobs:
         working-directory: ${{ github.workspace }}/barge
         run: |
           bash -x start_ocean.sh --no-dashboard 2>&1 --with-provider2 --with-rbac --with-c2d > start_ocean.log &
-          for i in $(seq 1 60); do
+          for i in $(seq 1 84); do
             sleep 5
             [ -f "$HOME/.ocean/ocean-contracts/artifacts/ready" -a -f "$HOME/.ocean/ocean-c2d/ready" ] && break
             done

--- a/.github/workflows/pytest.yml
+++ b/.github/workflows/pytest.yml
@@ -28,7 +28,7 @@ jobs:
         working-directory: ${{ github.workspace }}/barge
         run: |
           bash -x start_ocean.sh --no-dashboard 2>&1 --with-provider2 --with-rbac --with-c2d --with-thegraph > start_ocean.log &
-          for i in $(seq 1 60); do
+          for i in $(seq 1 84); do
             sleep 5
             [ -f "$HOME/.ocean/ocean-contracts/artifacts/ready" -a -f "$HOME/.ocean/ocean-c2d/ready" ] && break
             done

--- a/ocean_lib/models/data_nft.py
+++ b/ocean_lib/models/data_nft.py
@@ -6,12 +6,14 @@ import warnings
 from enum import IntEnum, IntFlag
 from typing import Optional
 
+from brownie import network
 from enforce_typing import enforce_types
 
 from ocean_lib.models.datatoken import Datatoken
-from ocean_lib.ocean.util import get_address_of_type
+from ocean_lib.ocean.util import create_checksum, get_address_of_type
 from ocean_lib.web3_internal.constants import ZERO_ADDRESS
 from ocean_lib.web3_internal.contract_base import ContractBase
+from ocean_lib.web3_internal.utils import check_network
 
 
 class DataNFTPermissions(IntEnum):
@@ -44,6 +46,11 @@ class DataNFT(ContractBase):
 
     def create_datatoken(self, datatoken_args, wallet) -> Datatoken:
         return datatoken_args.create_datatoken(self, wallet)
+
+    def calculate_did(self):
+        check_network(self.network)
+        chain_id = network.chain.id
+        return f"did:op:{create_checksum(self.address + str(chain_id))}"
 
 
 class DataNFTArguments:

--- a/ocean_lib/ocean/ocean_assets.py
+++ b/ocean_lib/ocean/ocean_assets.py
@@ -310,12 +310,11 @@ class OceanAssets:
         ddo = DDO()
 
         # Generate the did, add it to the ddo.
-        did = f"did:op:{create_checksum(data_nft.address + str(self._chain_id))}"
-        ddo.did = did
+        ddo.did = data_nft.calculate_did()
         # Check if it's already registered first!
-        if self._aquarius.ddo_exists(did):
+        if self._aquarius.ddo_exists(ddo.did):
             raise AquariusError(
-                f"Asset id {did} is already registered to another asset."
+                f"Asset id {ddo.did} is already registered to another asset."
             )
         ddo.chain_id = self._chain_id
         ddo.metadata = metadata
@@ -386,7 +385,7 @@ class OceanAssets:
 
         # Fetch the ddo on chain
         if wait_for_aqua:
-            ddo = self._aquarius.wait_for_ddo(did)
+            ddo = self._aquarius.wait_for_ddo(ddo.did)
 
         return (data_nft, datatokens, ddo)
 

